### ISSUE TITLE
Implement an ORM

### ISSUE THREAD

**Issue by wbprice** [Code4HR/okcandidate#7]
Bookshelf.js comes highly recommend and has a Hapi plugin.

http://bookshelfjs.org/
https://www.npmjs.com/package/hapi-shelf/

This will be easier to do once we have fixture data.

**Comment by wbprice** [comment]
Work is started on this on branch 7-implement-an-orm

**Comment by wbprice** [comment]
@ttavenner Do we need to generate a bunch of join tables or is the relationship between the tables implied in a different way?

**Comment by ttavenner** [comment]
The relationship is defined through foreign keys. You can see these in the database create script whenever a column references a table. i.e. `question.survey_id REFERENCES survey`. This means that it is a foreign key to `survey.id`.

Based on a quick perusal of the Bookshelf docs, it looks like my naming conventions should allow you to define these with implicit joins, i.e. this modified example should work:

```
let Survey = bookshelf.Model.extend({
  tableName: 'survey'
});

let Question = bookshelf.Model.extend({
  tableName: 'question',
  survey: function() {
    return this.belongsTo(Survey);
  }
});

Question.where({id: 1}).fetch({withRelated: ['survey']}).then(function(question) {
  console.log(JSON.stringify(question.related('survey')));
});
```

http://bookshelfjs.org/#Model-instance-belongsTo

**Comment by wbprice** [comment]
Resolved in #20.
